tests/unit/Scaffold/PathResolverTest.php

```
@@ -0,0 +1,275 @@
+            $result['root'],
+            'When the lock-recorded path is inside vendor, it must be honored.',
+        );
+        self::assertNull(
+            $result['warning'],
+            'No warning must be emitted when the lock-recorded provider path is valid and inside vendor.',
+        );
+    }
+
+    public function testResolveProviderRootUsesDefaultWhenLockEntryIsMissingPath(): void
+    {
+        $result = PathResolver::resolveProviderRoot($this->tempDir, 'pkg/name', null);
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\yii\scaffold\Scaffold\PathResolver' in method 'testResolveProviderRootUsesDefaultWhenLockEntryIsMissingPath'. (undefined)

(StaticAccess)